packages/financial-health-calculator/src/data-fetchers/npm.ts

```diff
@@ -0,0 +1,26 @@
+import pacote from 'pacote';
+import { scrapeNpm } from '../scrapers/npm.js';
+import type { NpmData } from '../types/npm/npm-data';
+
+type Funding = { url: string };
+
+function isFunding(element: unknown): element is Funding {
+  return element !== undefined && (element as Funding).url !== undefined;
+}
+
+export const fetchNpmData = async (packageName: string): Promise<NpmData> => {
+  const projectData = await pacote.manifest(packageName);
+  const scrapeResult = await scrapeNpm(packageName);
+
+  let fundingUrl = '';
+
+  if (isFunding(projectData.funding)) {
+    fundingUrl = projectData.funding.url;
+  }
+
+  return {
+    fundingUrl,
+    licenseIdentifier: scrapeResult.licenseIdentifier,
```

Comment:
Aren't the `licenseIdentifier` and `gitHubRepositoryIdentifier` available from the `projectData`? All information visible on NPM should come from the published `package.json`

(I feel that we already had this conversation, but could you refresh my memory?)

Reply:
If you include the fullMetadata option it also fetches the funding URL and license type:
```js
{
  name: 'eslint',
  version: '8.16.0',
  author: {
    name: 'Nicholas C. Zakas',
    email: 'nicholas+npm@nczconsulting.com'
  },
  description: 'An AST-based pattern checker for JavaScript.',
  bin: { eslint: 'bin/eslint.js' },
  main: './lib/api.js',
  exports: {
    './package.json': './package.json',
    '.': './lib/api.js',
    './use-at-your-own-risk': './lib/unsupported-api.js'
  },
  scripts: {
    ...scripts
  },
  gitHooks: { 'pre-commit': 'lint-staged' },
  'lint-staged': {
    '*.js': 'eslint --fix',
    '*.md': 'markdownlint --fix',
    'docs/src/rules/*.md': [
      'node tools/fetch-docs-links.js',
      'git add docs/src/_data/further_reading_links.json'
    ]
  },
  repository: { type: 'git', url: 'git+https://github.com/eslint/eslint.git' },
  funding: { url: 'https://opencollective.com/eslint' },
  homepage: 'https://eslint.org',
  bugs: { url: 'https://github.com/eslint/eslint/issues/' },
  dependencies: {
    ...deps
  },
  devDependencies: {
    ...devDeps
  },
  keywords: [ 'ast', 'lint', 'javascript', 'ecmascript', 'espree' ],
  license: 'MIT',
  engines: { node: '^12.22.0 || ^14.17.0 || >=16.0.0' },
  gitHead: '901ce0f1e32ea1e9e10ce4d8b37c0d750007a3c5',
  _id: 'eslint@8.16.0',
  _nodeVersion: '16.13.1',
  _npmVersion: '8.1.2',
  dist: {
    integrity: ...key
  },
  _npmUser: { name: 'eslintbot', email: 'nicholas+eslint@nczconsulting.com' },
  directories: {},
  maintainers: [
    { name: 'openjsfoundation', email: 'npm@openjsf.org' },
    { name: 'eslintbot', email: 'nicholas+eslint@nczconsulting.com' },
    { name: 'nzakas', email: 'nicholas@nczconsulting.com' },
    { name: 'ivolodin', email: 'ivolodin@gmail.com' }
  ],
  _npmOperationalInternal: {
    host: 's3://npm-registry-packages',
    tmp: 'tmp/eslint_8.16.0_1653086291946_0.7993754545335081'
  },
  _hasShrinkwrap: false,
  _resolved: 'https://registry.npmjs.org/eslint/-/eslint-8.16.0.tgz',
  _from: 'eslint@',
  _integrity: 'sha512-MBndsoXY/PeVTDJeWsYj7kLZ5hQpJOfMYLsF6LicLHQWbRDG19lK5jOix4DPl8yY4SUFcE3txy86OzFLWT+yoA==',
  _signatures: [...signatures
  ]
}
`

Reply:
I also see a `repository` property of which we could derive the repository page URL.

Reply:
Yes, I think we can

Reply:
![Do it](https://media.giphy.com/media/104ueR8J1OPM2s/giphy-downsized.gif)

Do it.

Reply:
NPM scraper is removed